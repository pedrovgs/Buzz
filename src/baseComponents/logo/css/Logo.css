.loading-screen-logo {
    width: 25%;
    height: 25%;
}