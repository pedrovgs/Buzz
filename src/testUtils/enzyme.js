import Enzyme from "enzyme";
import { createSerializer } from "enzyme-to-json";
import enzymeToJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
Enzyme.configure({ adapter: new Adapter() });

export function shallowComponentToJson(component) {
  return toJson(shallowComponent(component));
}

export function shallowComponent(component) {
  return Enzyme.shallow(component);
}

export function toJson(component) {
  return enzymeToJson(component);
}
