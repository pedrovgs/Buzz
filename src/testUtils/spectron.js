import { toMatchImageSnapshot } from "jest-image-snapshot";
import { Application } from "spectron";
import electron from "electron";
import path from "path";
import sharp from "sharp";

expect.extend({ toMatchImageSnapshot });

process.env.REACT_APP_RUNNING_TESTS = true;

jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;

let app;

export async function compareScreenshot() {
  await waitForReady();
  const screenshot = await app.client.saveScreenshot();
  const windowSize = await app.browserWindow.getSize();
  const resizedScreenshot = await sharp(screenshot)
    .resize(windowSize[0], windowSize[1])
    .toBuffer();
  expect(resizedScreenshot).toMatchImageSnapshot({
    failureThreshold: 5,
    failureThresholdType: "percent"
  });
}

export async function startApp() {
  app = new Application({
    path: electron,
    args: [path.join(__dirname, "..", "..", "electron-starter.js")],
    startTimeout: 10000
  });
  return app.start().then(() => {
    return waitForReady()
      .then(() => {
        return app.client.waitUntil(() => {
          return app.client
            .execute(() => {
              return window.store;
            })
            .then(store => {
              return typeof store !== "undefined";
            });
        });
      })
      .then(() => {
        return waitASec();
      })
      .then(() => {
        return resetAppState();
      })
      .then(() => {
        return app;
      });
  });
}

export async function stopApp() {
  await resetAppState();
  if (app) {
    return app.stop();
  } else {
    return Promise.resolve();
  }
}

export async function open(appRelativeUrl) {
  const appPath =
    "file:///" + path.join(__dirname, "..", "..", "build", "index.html");
  const applicationUrl = appPath + "#" + appRelativeUrl;
  return app.client.url(applicationUrl);
}

export async function getWindowTitle() {
  return app.client.getTitle();
}

export async function waitForVisible(selector) {
  return app.client.waitForVisible(selector);
}

export async function waitForInvisible(selector) {
  return app.client.waitForVisible(selector, true);
}

export async function waitASec() {
  return app.client.pause(1000);
}

export async function getUrlLoaded() {
  return app.client.getUrl();
}

export async function waitUntilUrlLoaded(expectedUrl) {
  return app.client.waitUntil(() => {
    return app.client.getUrl().then(url => {
      return url.endsWith(expectedUrl);
    });
  });
}

export function getClient() {
  return app.client;
}

export async function setAppState(state) {
  return getClient().execute(newState => {
    const action = {
      type: "RESET_STATE",
      state: newState
    };
    const store = window.store;
    store.dispatch(action);
    return store.getState();
  }, state);
}

export async function resetAppState() {
  return setAppState({});
}

export async function setValue(selector, value) {
  return getClient().setValue(selector, value);
}

async function waitForReady() {
  return app.client.waitUntilWindowLoaded();
}
