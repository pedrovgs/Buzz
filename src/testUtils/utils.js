export function isRunningTests() {
  const runningTestEnvVar = process.env.REACT_APP_RUNNING_TESTS;
  return runningTestEnvVar ? Boolean(runningTestEnvVar) === true : false;
}
