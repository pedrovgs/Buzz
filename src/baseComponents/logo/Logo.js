import React from "react";
import logo from "./images/logo.svg";
import "./css/Logo.css";

export default class Logo extends React.Component {
  render() {
    return (
      <img
        id="logo"
        src={logo}
        alt="Buzz logo"
        className="loading-screen-logo center"
      />
    );
  }
}
