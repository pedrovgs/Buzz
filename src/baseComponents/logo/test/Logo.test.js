import React from "react";
import Logo from "../Logo";
import { shallowComponent } from "../../../testUtils/enzyme";

describe("Logo", () => {
  it("shows the application logo", () => {
    const logo = shallowComponent(<Logo />);
    expect(logo).toMatchSnapshot();
  });
});
