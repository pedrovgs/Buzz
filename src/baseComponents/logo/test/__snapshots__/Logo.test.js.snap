// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo shows the application logo 1`] = `
<img
  alt="Buzz logo"
  id="logo"
  src="logo.svg"
  style={
    Object {
      "height": "40%",
      "padding": "12px",
      "width": "40%",
    }
  }
/>
`;
